tests/FsCodec.SystemTextJson.Tests/SerdesTests.fs

```diff
@@ -79,6 +80,42 @@ let [<Fact>] ``RejectNullStringConverter rejects null strings`` () =
     let value = { c = 1; d = null }
     raises<ArgumentNullException> <@ serdes.Serialize value @>
 
+type WithList = { x: int; y: list<int>; z: Set<int> }
+
+let [<Fact>] ``RejectNullConverter rejects null lists and Sets`` () =
+#if false // requires WithList to be CLIMutable, which would be a big imposition
+    let tir =
+        DefaultJsonTypeInfoResolver()
+            .WithAddedModifier(fun x ->
+                // if x.Kind <> JsonTypeInfoKind.Object then
+                for p in x.Properties do
+                    let pt = p.PropertyType
+                    if pt.IsGenericType && (let gtd = pt.GetGenericTypeDefinition() in gtd = typedefof<list<_>> || gtd = typedefof<Set<_>>) then
+                        p.IsRequired <- true)
+    let serdes = Options.Create(TypeInfoResolver = tir) |> Serdes
+    raises<exn> <@ serdes.Deserialize<WithList> """{"x":0}""" @>
+#else
+    let serdes = Options.Create(rejectNull = true) |> Serdes
+
+    // Fails with NRE when RejectNullConverter delegates to Default list<int> Converter
+    // seems akin to https://github.com/dotnet/runtime/issues/86483
+    // let res = serdes.Deserialize<WithList> """{"x":0,"y":[1]}"""
+    // test <@ [1] = res.y @>
+
+    // PROBLEM: Doesn't raise
+    raises<exn> <@ serdes.Deserialize<WithList> """{"x":0}""" @>
+    // PROBLEM: doesnt raise - there doesn't seem to be a way to intercept explicitly passed nulls
+    // raises<JsonException> <@ serdes.Deserialize<WithList> """{"x":0,"y":null}""" @>
+#endif
+
+#if false // I guess TypeShape is doing a reasaonable thing not propagating
+    // PROBLEM: TypeShape.Generic.exists does not call the predicate if the list or set is `null`
```

Comment:
My original workaround that came to mind was to walk the properties after the fact with TypeShape's Generic App, but I see I need to drop a level from that (as `null`s are not passed to the predicate, and I want to do a generic check in any case, so disregard this piece for now...